sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -33,11 +33,137 @@
 
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
+/**
+ * Table of forward branch Control Flow Integrity (CFI) counters.
+ *
+ * Columns: Name, Initital Value.
+ *
+ * Each counter is indexed by Name. The Initial Value is used to initialize the
+ * counters with unique values with a good hamming distance. The values are
+ * restricted to 12-bit to be able use immediate load instructions.
+ */
+// clang-format off
+#define ROM_CFI_COUNTERS_DEFINE(X) \
+  X(kCfiRomMain,    0xf05) \
+  X(kCfiRomInit,    0x042) \
+  X(kCfiRomVerify,  0x89d) \
+  X(kCfiRomTryBoot, 0x4c7) \
+  X(kCfiRomBoot,    0x0fb) \
+// clang-format on
+
+/**
+ * CFI counter initialization constants. These values are consumed in
+ * the `ROM_CFI_INIT()` macro.
+ */
+#define ROM_CFI_INIT_CONSTANTS_(name_, value_) name_##Val0 = value_,
+enum {
+  ROM_CFI_COUNTERS_DEFINE(ROM_CFI_INIT_CONSTANTS_)
+};
+
+/**
+ * CFI counters. Statically initialized with constant values.
+ */
+#define ROM_CFI_COUNTERS_TABLE_INIT_(name_, value_) value_,
+uint32_t rom_cfi_counters[] = {
+  ROM_CFI_COUNTERS_DEFINE(ROM_CFI_COUNTERS_TABLE_INIT_)
+};
+
+/**
+ * CFI counter indexes, derived from each `name_` positional value
+ * inside `ROM_CFI_COUNTERS_DEFINE()`.
+ */
+#define ROM_CFI_COUNTER_INDEXES_(name_, value_) name_,
+enum {
+  ROM_CFI_COUNTERS_DEFINE(ROM_CFI_COUNTER_INDEXES_)
+};
+
+enum {
+  // Counter increment constant.
+  kCfiRomInc = 0x5a,
```

Comment:
May want to call this `kCfiInc` or `kCfiIncrement` unless you want to use different increments for rom and rom_ext.

Reply:
Done.